packages/my-extended-retail-app/pwa-kit-overrides/app/components/footer/index.jsx

```diff
@@ -0,0 +1,275 @@
+                                {...otherProps}
+                            >
+                                <Select
+                                    value={locale}
+                                    onChange={({target}) => {
+                                        setLocale(target.value)
+
+                                        // Update the `locale` in the URL.
+                                        const newUrl = getPathWithLocale(target.value, buildUrl, {
+                                            disallowParams: ['refine']
+                                        })
+
+                                        window.location = newUrl
+                                    }}
+                                    variant="filled"
+                                    {...styles.localeDropdown}
+                                >
+                                    {supportedLocaleIds.map((locale) => (
+                                        <LocaleText
+                                            as="option"
+                                            value={locale}
+                                            shortCode={locale}
+                                            key={locale}
+                                            {...styles.localeDropdownOption}
+                                        />
+                                    ))}
+                                </Select>
+                            </FormControl>
+                        </Box>
+                    )}
+
+                    <Divider {...styles.horizontalRule} />
+
+                    <Box {...styles.bottomHalf}>
+                        <Text {...styles.copyright}>
+                            &copy; {new Date().getFullYear()}{' '}
+                            {intl.formatMessage({
+                                id: 'footer.message.copyright',
+                                defaultMessage:
+                                    'Salesforce or its affiliates. All rights reserved. This is a demo store only. Orders made WILL NOT be processed.'
+                            })}
+                        </Text>
+
+                        <HideOnDesktop>
+                            <LegalLinks variant="vertical" />
+                        </HideOnDesktop>
+                        <HideOnMobile>
+                            <LegalLinks variant="horizontal" />
+                        </HideOnMobile>
+                    </Box>
+                </StylesProvider>
+            </Box>
+        </Box>
+    )
+}
+
+export default Footer
+
+const Subscribe = ({...otherProps}) => {
+    const styles = useStyles()
+    const intl = useIntl()
+
+    const einstein = useEinstein()
+    const analytics = useSomeAnalytics()
+
+    const onClick = () => {
+        einstein.sendSomeEvent()
+        analytics.sendSomeEvent()
+
+        // Send custom event for Google Analytics
+        window.gtag('event', 'email_signup')
```

Comment:
This is the main change for this footer file. But unfortunately I had to override the file by replacing everything. Ideally I would need to override just the `Subscribe` component and not the entire `Footer`.

More breaking down of components is needed for the retail-react-app template.